vm-executor-wasmer/src/wasmer_imports.rs

```diff
@@ -12,1312 +12,1312 @@ use crate::wasmer_vm_hooks::VMHooksWrapper;
 
 #[rustfmt::skip]
 fn wasmer_import_get_gas_left(env: &VMHooksWrapper) -> i64 {
-    env.vm_hooks.get_gas_left()
+    env.vm_hooks.borrow_mut().get_gas_left()
```

Comment:
This implementation will change, to avoid needing to re-generate each time, let's encapsulate into a method that produces a `&mut VMHooks` in `VMHooksWrapper`.

I think you need to provide it in continuation-passing style, `with_vm_hooks`.